tsl/src/continuous_aggs/invalidation.c

```diff
@@ -314,27 +314,48 @@ invalidation_hyper_log_add_entry(int32 hyper_id, int64 start, int64 end)
  * hypertable.
  */
 void
-invalidation_add_entry(const Hypertable *ht, ContinuousAggHypertableStatus caggstatus,
-					   int32 entry_id, int64 start, int64 end)
+continuous_agg_invalidate_raw_ht(const Hypertable *raw_ht, int64 start, int64 end)
 {
-	Assert(ht != NULL);
+	continuous_agg_invalidate_mat_ht(raw_ht, NULL, start, end);
+}
```

Comment:
It is odd that you split it up into two functions, but then you put all the logic in one of the functions and use `NULL` indicate the non-materialized table case.

I would have expected the "remove invalidation" function to be split up as well and used, e.g., something like this:

```
void
continuous_agg_invalidate_raw_ht(const Hypertable *raw_ht, int64 start, int64 end)
{
  if (hypertable_is_distributed(raw_ht)) {
    remote_invalidation_log_add_entry(INVALIDATION_HYPER_LOG_ADD_ENTRY_FUNCNAME, raw_ht->fd.id, start, end);
  } else {
    invalidation_hyper_log_add_entry(raw_ht->fd.id, start, end); 
  }
}
```